client/src/containers/GenerateScreen.js

```diff
@@ -0,0 +1,319 @@
+    marginRight: '10px',
+  },
+  casino: {
+    marginRight: '5px',
+  },
+};
+
+class GenerateScreen extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      step: 1,
+      selectedPlayersKeys: [],
+      extraPlayersSelectedOption: null,
+      numberOfPlayersPerTeam: 1,
+      teams: null,
+    };
+    this.handleToggleCheckbox = this.handleToggleCheckbox.bind(this);
+    this.handleMoveToAnotherStep = this.handleMoveToAnotherStep.bind(this);
+    this.handleRadioChange = this.handleRadioChange.bind(this);
+    this.handleChangeNumberOfPlayersPerTeam = this.handleChangeNumberOfPlayersPerTeam.bind(this);
+    this.getStepAttributes = this.getStepAttributes.bind(this);
+    this.getStepOneAttributes = this.getStepOneAttributes.bind(this);
+    this.getStepTwoAttributes = this.getStepTwoAttributes.bind(this);
+    this.getStepThreeAttributes = this.getStepThreeAttributes.bind(this);
+    this.handleGenerateNewTeams = this.handleGenerateNewTeams.bind(this);
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (
+      this.state.step === 2 &&
+      this.state.numberOfPlayersPerTeam > this.state.selectedPlayersKeys.length
+    ) {
+      this.setState({ numberOfPlayersPerTeam: 1 });
+    }
+
+    if (this.state.step === 3 && !this.state.teams) {
+      this.handleGenerateNewTeams();
+    }
+
+    if (this.state.step === 3 && prevState.step !== 3 ) {
+      this.handleGenerateNewTeams();
+    }
+  }
+
+  handleGenerateNewTeams() {
+    const selectedPlayers = this.props.playersArray.filter(p => this.state.selectedPlayersKeys.includes(p.id));
+    this.setState({ teams: generateRandomTeam(selectedPlayers, this.state.numberOfPlayersPerTeam)})
+  }
+
+  getStepAttributes(step) {
+    switch (step) {
+      case 1:
+        return this.getStepOneAttributes();
+      case 2:
+        return this.getStepTwoAttributes();
+      case 3:
+        return this.getStepThreeAttributes();
+      default:
+        break;
+    }
+  }
+
+  getStepOneAttributes() {
+    const { classes } = this.props;
+
+    const content = (
+      <Grid container spacing={24} className={classes.content}>
+        <Grid item xs={12}>
+          <Typography variant="body1">
+            Pick at least one player to go to the next step
+          </Typography>
+          <Typography variant="caption">
+            {this.state.selectedPlayersKeys.length} of {this.props.playersArray.length} players selected
+          </Typography>
+        </Grid>
+        <Grid item xs={12} className={classes.playerList}>
+          <List>
+            {this.props.playersArray.map(player =>
+              this.renderPlayerListItem(player.id, player.name, player.ratings, this.state.selectedPlayersKeys.includes(player.id)))
+            }
+          </List>
+        </Grid>
+      </Grid>
+    );
+
+    return {
+      content,
+      title: 'Select Players',
+      leftButtonLabel: 'Back',
+      rightButtonDisabled: this.state.selectedPlayersKeys <= 0,
```

Comment:
<= 0 artinya apa glo? nggk pernah liat gw

Reply:
'less than or equal' wkakakaka. Itu harusnya `(this.state.selectedPlayersKeys <= 0)` which returns true or false buat tentuin kalo rightButtonsnya harus didisable atau ngga